src/org/exist/collections/Collection.java

```diff
@@ -2305,4 +2303,32 @@ public QNameRangeIndexSpec getIndexByQNameConfiguration(final DBBroker broker, f
         final IndexSpec idxSpec = getIndexConfiguration(broker);
         return (idxSpec == null) ? null : idxSpec.getIndexByQName(qname);
     }
+
+    @Override
+    public ResourceMetadata getMetadata() {
+        return new CollectionMetadata();
```

Comment:
This does not need to be done on each `get`, it would be better to follow the model set out in `DocumentImpl#getMetadata`, i.e. the metadata only needs to be read once or after it is updated.